api/clients/v2/coretypes/eigenda_cert.go

```diff
@@ -73,6 +74,19 @@ type EigenDACert interface {
 	isEigenDACert()
 }
 
+// DeserializeEigenDACert deserializes raw bytes into an EigenDACert
+// based on the provided version and serialization type
+func DeserializeEigenDACert(data []byte, version CertificateVersion, ct CertSerializationType) (EigenDACert, error) {
```

Comment:
do we need this function? Can't we just create 2 separate `DeserializeABIEncodedCert` and `DeserializeRLPEncodedCert`?

Reply:
EDIT: hmm looking at the code in https://github.com/Layr-Labs/eigenda/pull/2246/files realized this might be harder than I thought. Because the store IS actually multiplexed and already doesn't know whether its in an arb or op context...

Gosh I really hate our store abstraction. I guess we're stuck with it for now but we really need to rework it after we nuke V1. Prob worth pondering a little how we got into this state, to make sure we don't ever end up in similar state in other projects. My main issue with it is this dynamic multiplexing. You should basically as much as possible never have such a god class in teh middle of the code. Better to wire things independently and keep information as local as possible.